src/main/java/io/pravega/connectors/hadoop/PravegaInputRecordReader.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017 Dell/EMC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pravega.connectors.hadoop;
+
+import io.pravega.client.ClientFactory;
+import io.pravega.client.batch.BatchClient;
+import io.pravega.client.batch.SegmentIterator;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.impl.JavaSerializer;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * A RecordReader that can read events from an InputSplit(segment in pravega).
+ */
+public class PravegaInputRecordReader<V> extends RecordReader<MetadataWritable, V> {
+
+    private static final Logger log = LoggerFactory.getLogger(PravegaInputRecordReader.class);
```

Comment:
This is not consistent with pravega/pravega's checkstyle rules. Perhaps it would be worth bringing over its targets for checkstyle and findbugs to do validation

Reply:
I checked the rules, but don't see the naming breaks any rule. can you point out which part is not consistent?